src/neural/cuda/layers.cc

```diff
@@ -439,10 +439,12 @@ template <>
 void SELayer<half>::Eval(int N, half* output, const half* input,
                          const half* input2, void* scratch, size_t scratch_size,
                          cudnnHandle_t /*cudnn*/, cublasHandle_t cublas) {
+  bool se_done = false;
   if (kUseFusedSELayer) {
-    Se_Fp16_NHWC(N, C, numFc1Out_, output, input2, input, w1_, b1_, w2_, b2_,
-                 bPrev_);
-  } else {
+    se_done = Se_Fp16_NHWC(N, C, numFc1Out_, output, input2, input, w1_, b1_,
+                           w2_, b2_, bPrev_);
+  } 
+  if (!se_done) {
```

Comment:
I'm sorry, but why is this not one if statement with and "and"? Would be much more legible...

Reply:
Do you mean something like:
```
if (!kUseFusedSELayer || !Se_Fp16_NHWC(N, C, numFc1Out_, output, input2, input, w1_, b1_, w2_, b2_, bPrev_) ) {
   // handle non-fused case
}
```
Personally I find it less legible, but I don't have any strong objection.